kotlin/configure-subnetwork-trace/src/main/java/com/esri/arcgisruntime/sample/configuresubnetworktrace/MainActivity.kt

```diff
@@ -0,0 +1,340 @@
+      // update the list of coded values
+      values = codedValueDomain.codedValues
+      // show the values spinner
+      setVisible(valuesBackgroundView.id)
+      // update the values spinner adapter
+      valuesSpinner.adapter = ArrayAdapter<String>(
+        applicationContext,
+        android.R.layout.simple_spinner_item,
+        // add the the coded values from the coded value domain to the values spinner
+        codedValueDomain.codedValues.map { it.name }
+      )
+      // if the domain is not a coded value domain
+    } ?: when (attribute.dataType) {
+      UtilityNetworkAttribute.DataType.BOOLEAN -> {
+        setVisible(valueBooleanButton.id)
+      }
+      UtilityNetworkAttribute.DataType.DOUBLE, UtilityNetworkAttribute.DataType.FLOAT -> {
+        // show the edit text and only allow numbers (decimals allowed)
+        valuesEditText.inputType =
+          InputType.TYPE_CLASS_NUMBER or InputType.TYPE_NUMBER_FLAG_DECIMAL
+        setVisible(valuesEditText.id)
+      }
+      UtilityNetworkAttribute.DataType.INTEGER -> {
+        // show the edit text only allowing for integer input
+        valuesEditText.inputType = InputType.TYPE_CLASS_NUMBER
+        setVisible(valuesEditText.id)
+      }
+      else -> {
+        Log.e(TAG, "Unexpected utility network attribute data type.")
+      }
+
+    }
+  }
+
+  /**
+   * Show the given UI view and hide the others which share the same space.
+   */
+  private fun setVisible(id: Int) {
+    when (id) {
+      valuesBackgroundView.id -> {
+        valuesBackgroundView.visibility = View.VISIBLE
+        valueBooleanButton.visibility = View.INVISIBLE
+        valuesEditText.visibility = View.INVISIBLE
+      }
+      valuesEditText.id -> {
+        valuesEditText.visibility = View.VISIBLE
+        valueBooleanButton.visibility = View.INVISIBLE
+        valuesBackgroundView.visibility = View.INVISIBLE
+      }
+      valueBooleanButton.id -> {
+        valueBooleanButton.visibility = View.VISIBLE
+        valuesBackgroundView.visibility = View.INVISIBLE
+        valuesEditText.visibility = View.INVISIBLE
+      }
+    }
+  }
+
+  /**
+   * Add a new barrier condition to the trace options.
+   */
+  fun addCondition(view: View) {
+    // if source tier doesn't contain a trace configuration, create one
+    val traceConfiguration = sourceTier?.traceConfiguration ?: UtilityTraceConfiguration().apply {
+      // if the trace configuration doesn't contain traversability, create one
+      traversability ?: UtilityTraversability()
+    }
+
+    // NOTE: You may also create a UtilityCategoryComparison with UtilityNetworkDefinition.Categories and UtilityCategoryComparisonOperator
+    sources?.get(sourceSpinner.selectedItemPosition)?.let { attribute ->
+      operators?.get(operatorSpinner.selectedItemPosition)?.let { attributeOperator ->
+        // NOTE: You may also create a UtilityCategoryComparison with UtilityNetworkDefinition.Categories and UtilityCategoryComparisonOperator
```

Comment:
duplicate comments here: though can you help me understand why we are telling the user this at this point in the code? From the comment I was expecting the code below to show a UtilityCategoryComparison being created, though I don't see that happening.

Reply:
Hmm yeah, this is from the common design--I've just scrapped it. It's confusing to me too